examples/dbt-on-cloud-composer/basic/dbt-project/cloudbuild.yaml

```diff
@@ -37,27 +38,47 @@ steps:
     ]
 
 # This is the dbt service account key to run the dbt docs command
-# This step copy the key that stored in a GCS bucket to cloud build VM
-# TODO : This approach could be reviewed for better security
+# Set the _DBT_SERVICE_ACCOUNT to your dbt service account
+# Example format : sa-dbt@project-id.iam.gserviceaccount.com
 
-- name: gcr.io/cloud-builders/gsutil
+- name: gcr.io/cloud-builders/gcloud
   id: Get SA key
+  entrypoint: "bash"
   args:
-  - cp
-  - gs://pso-dbt-airflow-demo-bucket/key/pso-dbt-airflow-demo-a87a347b6390.json
-  - key/pso-dbt-airflow-demo-a87a347b6390.json
+  - "-c"
+  - |
+    gcloud iam service-accounts keys create sa-key.json --iam-account=${_DBT_SERVICE_ACCOUNT}
```

Comment:
Is it essential to create a key for the service account to use this example? We recommend not exporting keys, and using impersonation instead.

Reply:
hi @iht , in this example the service account key is needed to run dbt CLI.
The [CLI is accepting SA key file](https://docs.getdbt.com/reference/warehouse-profiles/bigquery-profile#service-account-file)

Reply:
Hi, I pushed another commit. I just knew dbt has option to directly use "impersonate_service_account" parameter in the run.